File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        int n = people.size(), i = 0, j = n - 1, ans = 0;
        
        while(i <= j){
            int sum = people[i] + people[j];
            if(sum <= limit){
                ++i;
                --j;
            }else if(sum > limit){
                --j;
            }
            ++ans;
        }
        return ans;
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int l = 0, r = people.length - 1;
        int res = 0;
        Arrays.sort(people);
        
        while(l <= r){
            int wt = people[l] + people[r];
            if(wt <= limit){
                ++l;
            }
            --r;
            ++res;
        }
        
        return res;
    }
}